### PR TITLE
feat: add close button to adjudicate notification

### LINKED ISSUE #559: [UST-249] [Frontend] Users should be able to close the floating button of adjudication
**Is your feature request related to a problem? Please describe.**
Currently, user cannot hide the floating button on the screen. User should be able to close the floating button if they don't want to adjudicate the report cases.

**Describe the solution you'd like**
- Add a close button on the adjudication floating button.
- Popup the confirm dialog when click the close button.
- Hide the floating button if confirmed, but still show the floating button if there is a new report to adjudicate.

**Additional context**
[Figma](https://www.figma.com/design/WFpSKfe9741XSlXkOXz8s0/Unirep-Social?node-id=3878-34326&t=9k5mEaftlWOoNzQe-4)

--- 
Story point: 3
Reporter: @Xiawpohr

### PR DESCRIPTION
**Pull request by chen-yanlong**
## Summary

add close button to adjudicate notification

## Linked Issue

close #559 

## Details
- Add a close button on the adjudication floating button.
- Popup the confirm dialog when click the close button.
- Hide the floating button if confirmed, but still show the floating button if there is a new report to adjudicate.

## Impacted Areas

packages/frontend/src/features/reporting/components/AdjudicateNotification

## Verification Steps

close button of notification
1. press close button of the adjudication notification
2. should pop out an dialog of confirmation

confirmation
4. close if pressed confirmed(also closes the notification)
5. jump to adjudication if pressed cancel(also closes the notification)
6. close if pressed close button(does not closes the notification)

### CODE REVIEW

File: packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx [hunk 1]
```diff
@@ -75,25 +77,63 @@ function useActiveAdjudication() {
...
+
+    const closeAdjudicate = () => {
         toggle(false)
     }
 
+    const closeConfirm = () => {
+        setConfirmOpen(false)
+    }
+
+    const handleCloseClick = () => {
```
Comment (Xiawpohr) [inline packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx:98]: Suggest to rename this function to `openConfirmation` for clarity.

File: packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx [hunk 2]
```diff
@@ -75,25 +77,63 @@ function useActiveAdjudication() {
...
+        if (activeAdjudication) {
+            setButtonVisible(true)
+        }
+    }, [activeAdjudication])
+
+    const closeAdjudicate = () => {
         toggle(false)
     }
 
+    const closeConfirm = () => {
```
Comment (Xiawpohr) [inline packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx:94]: Suggest to rename this function to `closeConfirmation` for clarity.

File: packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx [hunk 3]
```diff
@@ -75,25 +77,63 @@ function useActiveAdjudication() {
...
+        setConfirmOpen(true)
+    }
+
+    const goAdjudicate = () => {
+        toggle(true)
+        setButtonVisible(false)
+        setConfirmOpen(false)
+    }
+
+    const confirmClose = () => {
```
Comment (Xiawpohr) [inline packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx:108]: Suggest to rename this function to `rejectAdjudication` for clarity.

File: packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx [hunk 4]
```diff
@@ -75,25 +77,63 @@ function useActiveAdjudication() {
...
 
+    const closeConfirm = () => {
+        setConfirmOpen(false)
+    }
+
+    const handleCloseClick = () => {
+        setConfirmOpen(true)
+    }
+
+    const goAdjudicate = () => {
```
Comment (Xiawpohr) [inline packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx:102]: Suggest to rename this function to `openAdjudication` for clarity.

File: packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx [hunk 5]
```diff
@@ -75,25 +77,63 @@ function useActiveAdjudication() {
...
 
-    const onClose = () => {
-        refetch()
+    useEffect(() => {
+        if (activeAdjudication) {
+            setButtonVisible(true)
+        }
+    }, [activeAdjudication])
+
+    const closeAdjudicate = () => {
```
Comment (Xiawpohr) [inline packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx:90]: Suggest to rename this function to `closeAdjudication` for clarity.

File: packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx [hunk 6]
```diff
@@ -75,25 +77,63 @@ function useActiveAdjudication() {
...
         toggle(false)
     }
```
Comment (Xiawpohr) [inline packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx:92]: The button should be visible again when closing adjudication dialog.

File: packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx (hunk 1 above)
Comment (chen-yanlong) [reply packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx]: fixed

File: packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx (hunk 2 above)
Comment (chen-yanlong) [reply packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx]: fixed

File: packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx (hunk 3 above)
Comment (chen-yanlong) [reply packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx]: fixed

File: packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx (hunk 4 above)
Comment (chen-yanlong) [reply packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx]: fixed

File: packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx (hunk 5 above)
Comment (chen-yanlong) [reply packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx]: fixed

File: packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx (hunk 6 above)
Comment (chen-yanlong) [reply packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx:92]: The button's visibility depends on `buttonVisible`, so it should be visible again when closing adjudication dialog now. Please let me know if i misunderstood.

Comment (Xiawpohr) [reply packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx:92]: The floating button becomes invisible if I close adjudication dialog. You should make it visible again.

https://github.com/user-attachments/assets/d5d3c21f-288c-43e9-b56e-a5bc7b433dc9

Comment (chen-yanlong) [reply packages/frontend/src/features/reporting/components/AdjudicateNotification/AdjudicateNotification.tsx:94]: Got it ! Solved.

**Comment by Xiawpohr** [comment]
@chen-yanlong Please resolve the frontend build error.

**Comment by chen-yanlong** [comment]
@Xiawpohr frontend build error fixed, please have a look.

**Review by Xiawpohr — APPROVED** [review]
Thanks for your contribution.

**Pull request merged**